builder/processor/src/main/java/io/helidon/builder/processor/GenerateAbstractBuilder.java

```diff
@@ -36,232 +49,182 @@
+                                             .from(TypeName.create(PROTOTYPE_CONFIGURED_BUILDER))
+                                             .addTypeArgument(TypeName.createFromGenericDeclaration("BUILDER"))
+                                             .addTypeArgument(TypeName.createFromGenericDeclaration("PROTOTYPE"))
+                                             .build());
+            } else {
+                builder.addInterface(TypeName.builder()
+                                             .from(TypeName.create(PROTOTYPE_BUILDER))
+                                             .addTypeArgument(TypeName.createFromGenericDeclaration("BUILDER"))
+                                             .addTypeArgument(TypeName.createFromGenericDeclaration("PROTOTYPE"))
+                                             .build());
             }
-        }
 
-        pw.print(SOURCE_SPACING);
-        pw.print(SOURCE_SPACING);
-        pw.println("}");
-        pw.println();
+            fields(builder, typeContext, true);
 
-        // method "from(prototype)"
-        fromInstanceMethod(pw, typeContext, prototypeWithTypes);
-        fromBuilderMethod(pw, typeContext, typeArgumentNames);
+            // method "from(prototype)"
+            fromInstanceMethod(builder, typeContext, prototype);
+            fromBuilderMethod(builder, typeContext, typeArguments);
 
-        // method preBuildPrototype() - handles providers, decorator
-        preBuildPrototypeMethod(pw, typeContext);
-        validatePrototypeMethod(pw, typeContext);
+            // method preBuildPrototype() - handles providers, decorator
+            preBuildPrototypeMethod(builder, typeContext);
+            validatePrototypeMethod(builder, typeContext);
 
-        CustomMethods customMethods = typeContext.customMethods();
+            //custom method adding
+            addCustomBuilderMethods(typeContext, builder);
 
-        for (CustomMethods.CustomMethod customMethod : customMethods.builderMethods()) {
-            // builder specific custom methods (not part of interface)
+            // setters and getters of builder
+            builderMethods(builder, typeContext);
+            toString(builder,
+                     typeContext,
+                     prototype.className() + "Builder",
+                     superType.isPresent(),
+                     typeContext.customMethods().prototypeMethods(),
+                     true);
+
+            // before the builder class is finished, we also generate a protected implementation
+            generatePrototypeImpl(builder, typeContext, typeArguments);
+        });
+    }
+
+    private static void addCustomPrototypeMethods(TypeContext typeContext, InnerClass.Builder builder) {
+        for (CustomMethods.CustomMethod customMethod : typeContext.customMethods().prototypeMethods()) {
+            // TODO these sections should be moved to CustomMethod implementation once we have class model
```

Comment:
This TODO should be replaced with an issue ref

Reply:
The whole method removed since it is no longer needed.